### PR TITLE
Sprk 269

### PR DESCRIPTION
**Pull request by thijssnelleman**
Major refactor of parallel portfolio.

### CODE REVIEW

**Review by jeroenrook — CHANGES_REQUESTED** [review]

File: sparkle/platform/settings_help.py [hunk 1]
```diff
@@ -90,8 +72,7 @@ class Settings:
 
     DEFAULT_ablation_racing = False
 
-    DEFAULT_paraport_overwriting = False
-    DEFAULT_paraport_process_monitoring = ProcessMonitoring.REALISTIC
+    DEFAULT_parallel_portfolio_check_interval = 4
```
[the remark this answers is not in the document]
Comment (thijssnelleman) [reply sparkle/platform/settings_help.py:75]: Value as found in the original parallel portfolio implementation

File: sparkle/solver/solver.py [hunk 2]
```diff
@@ -114,7 +114,7 @@ def is_deterministic(self: Solver) -> str:
 
         return deterministic
 
-    def build_solver_cmd(self: Solver, instance: str, configuration: dict = None,
+    def build_solver_cmd(self: Solver, instance: str, configuration: dict = {},
```
[the remark this answers is not in the document]
Comment (thijssnelleman) [reply sparkle/solver/solver.py:117]: This is a very good point. This happens in many locations of Sparkle and should be checked

File: CLI/run_parallel_portfolio.py [hunk 3]
```diff
@@ -0,0 +1,291 @@
...
+                run_status_path / f"{solver.name}_{instance.name}_{log_timestamp}.log"
+            runsolver_values_log =\
+                run_status_path / f"{solver.name}_{instance.name}_{log_timestamp}.var"
+            raw_result_path =\
+                run_status_path / f"{solver.name}_{instance.name}_{log_timestamp}.raw"
+            print(runsolver_values_log)
+            solver_call_list = solver.build_solver_cmd(
+                str(instance),
+                configuration={"specifics": "",
+                               "seed": 1,
```
[the remark this answers is not in the document]
Comment (thijssnelleman) [reply CLI/run_parallel_portfolio.py]: This has now been fixed: The user can specify in the settings how many random seeds must be made for each solver, and can run multiple solvers at the same time in one portfolio.

**Comment by jeroenrook** [comment]
Running the example `parallel_portfolio_runtime.sh` shows an PAR10 score of the portfolio of 4817.01, which cannot be possible with a cutoff time of 60 seconds.

**Comment by jeroenrook** [comment]
What is the purpose of cancelled and best solver for the portfolio table in the report? Als tables are missing captions and are not centred.

**Review by jeroenrook — CHANGES_REQUESTED** [review]
The reporting of the portfolio seems to not work as desired. See comments above. For a portfolio I expect to see the solving time (by running solvers in parallel) and the CPU time. For example, Solver A takes 5 seconds to solve and Solver B is cancelled after 5 seconds, then the solving time would be 5 seconds, but the CPU time 10.

**Comment by jeroenrook** [comment]
Report is also written to Output/Selection, which should be changed to ParallelPortfolio

**Pull request merged**